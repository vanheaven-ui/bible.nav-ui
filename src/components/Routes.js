import { useState } from 'react';
import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
import { LastLocationProvider } from 'react-router-last-location';
import App from './App';
import Book from './Book';
import Footer from './common/Footer';
import Navbar from './common/Navbar';
import LoggedInNavbar from './common/NavbarLogin';
import Login from './Login';
import Signup from './Signup';
import Verse from './Verse';
import FavoritesList from '../containers/FavoritesList';

const Routes = () => {
  const [loggedIn, setLoggedIn] = useState(false);

  const handleLogin = () => {
    setLoggedIn(!loggedIn);
  };

  return (
    <Router>
      <LastLocationProvider>
        <div className="dark-overlay" />
        <header>
          { loggedIn && <LoggedInNavbar status={handleLogin} /> }
          { !loggedIn && <Navbar /> }
        </header>
        <Switch>
          <Route exact path="/">
            <App />
          </Route>
          <Route path="/signup">
            <Signup />
          </Route>
          <Route path="/login">
            <Login update={handleLogin} />
          </Route>
          <Route exact path="/books/:id">
            <Book login={loggedIn} />
          </Route>
          <Route path="/books/:id/verses/:verse">
            <Verse />
          </Route>
          <Route path="/favorites">
            <FavoritesList />
          </Route>
        </Switch>
        <Footer />
      </LastLocationProvider>
    </Router>
  );
};

export default Routes;
