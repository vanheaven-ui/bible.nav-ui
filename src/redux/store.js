import { applyMiddleware, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';
import thunkMiddleware from 'redux-thunk';
import rootReducer from './reducers';

const middlewareEnhancer = applyMiddleware(thunkMiddleware);
const composedEnhancer = composeWithDevTools(middlewareEnhancer);

const store = createStore(rootReducer, undefined, composedEnhancer);

export default store;
