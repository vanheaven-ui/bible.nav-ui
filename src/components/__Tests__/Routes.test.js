import React from 'react';
import ReactDOM from 'react-dom';
import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import rootDiv from '../../testHelper';
import '@testing-library/jest-dom';
import Routes from '../Routes';
import store from '../../redux/store';

describe('Routes Component', () => {
  beforeAll(() => {
    global.Date.now = jest.fn(() => 1620718496066);
  });

  it('renders without crashing', () => {
    ReactDOM.render(<Provider store={store}><Routes /></Provider>, rootDiv());
  });
  
  it('does not change unexpectedly', () => {
    const tree = renderer.create(
      <Provider store={store}><Routes /></Provider>,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });  
});
