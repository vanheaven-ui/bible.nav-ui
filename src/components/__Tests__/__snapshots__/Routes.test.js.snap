// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes Component does not change unexpectedly 1`] = `
Array [
  <div
    className="dark-overlay"
  />,
  <header>
    <nav
      className="navbar"
    >
      <div
        className="nav-container"
      >
        <a
          className="brand nav-logo"
          href="/"
          onClick={[Function]}
        >
          <img
            alt="logo"
            height="50px"
            src="logo1.png"
            width="50px"
          />
          Bible.nav
        </a>
        <ul
          className="nav-menu"
        >
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-links"
              href="/signup"
              onClick={[Function]}
            >
              Signup
            </a>
          </li>
          <li
            className="nav-item"
          >
            <a
              aria-current={null}
              className="nav-links"
              href="/login"
              onClick={[Function]}
            >
              Login
            </a>
          </li>
        </ul>
        <button
          className="nav-icon"
          onClick={[Function]}
          type="button"
        >
          <i
            aria-label="times"
            className="fas fa-bars"
          />
        </button>
      </div>
    </nav>
  </header>,
  <div
    className="App"
  >
    <section
      className="hero"
    >
      <div
        className="hero-text"
      >
        <div
          className="container"
        >
          <div
            className="row"
          >
            <div
              className="col-md-6 col-12 order-md-first order-last"
              id="left-col"
            >
              <div
                className="left-hero"
              >
                <h2>
                   Navigating the Bible simplified
                </h2>
                <p>
                  Your favorite bible verse is a login away
                </p>
                <button
                  className="orange-button"
                  type="button"
                >
                  <a
                    href="/signup"
                    onClick={[Function]}
                  >
                    START HERE
                  </a>
                </button>
              </div>
            </div>
            <div
              className="col-md-6 col-12 order-md-last order-first"
            >
              <div
                className="hero-heading"
              >
                <small>
                  Tuesday, May 11, 2021
                </small>
                <h4>
                  VERSE OF THE DAY:
                </h4>
                <h2>
                  
                </h2>
              </div>
              <div
                className="verse-section"
              >
                <p
                  className="verse-of-day"
                >
                  
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
    <section
      className="book-list"
    >
      <h2>
        GET BOOKS OF THE BIBLE
      </h2>
      <div
        className="container-fluid"
        id="bible-sections"
      >
        <div
          className="row"
        >
          <div
            className="col-md-6 col-12"
            id="navigate"
          >
            <h4>
              Let us navigate the sections of the bible
            </h4>
            <i
              className="fas fa-caret-right fa-3x right-fa"
            />
            <i
              className="fas fa-caret-down fa-3x down-fa"
            />
          </div>
          <div
            className="col-md-6 col-12"
          >
            <button
              name="new"
              onClick={[Function]}
              type="button"
            >
              NEW TESTAMENT
            </button>
            <button
              name="old"
              onClick={[Function]}
              type="button"
            >
              OLD TESTAMENT
            </button>
          </div>
        </div>
      </div>
    </section>
    <section
      className="favorites-pitch"
    >
      <h3>
        CREATE YOUR FAVORITES
      </h3>
      <p>
        Have your favorites in one place by registering or signing in
      </p>
      <div
        className="buttons"
      >
        <button
          className="orange-button"
          type="button"
        >
          <a
            href="/signup"
            onClick={[Function]}
          >
            REGISTER
          </a>
        </button>
        <button
          className="outline-btn"
          type="button"
        >
          <a
            href="/login"
            onClick={[Function]}
          >
            LOGIN
          </a>
        </button>
      </div>
    </section>
  </div>,
  <footer
    id="footer"
  >
    <div
      className="footer-wrap"
    >
      <h3
        className="footer-logo"
      >
        <img
          alt="logo"
          height="50px"
          src="logo1.png"
          width="50px"
        />
        <a
          href="/"
          onClick={[Function]}
        >
          Bible.nav
        </a>
      </h3>
      <p
        className="address"
      >
        jdkfljsdfjdslfds
      </p>
      <ul
        className="social-links"
      >
        <li>
          <a
            href="https://www.linkedin.com/in/vanheaven/"
          >
            <i
              className="fab fa-linkedin"
            />
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/MworekwaE"
          >
            <i
              className="fab fa-twitter-square"
            />
          </a>
        </li>
        <li>
          <a
            href="https://github.com/vanheaven-ui"
          >
            <i
              className="fab fa-github-square"
            />
          </a>
        </li>
      </ul>
      <p
        className="copyright"
      >
        Bible.nav 2021. All rights reserved
      </p>
    </div>
  </footer>,
]
`;
